### DISCUSSION TITLE
Access context value in service

### DISCUSSION THREAD

**Discussion by Jonur** [statelyai/xstate#3903] [Q&A]
Hello, we have a machine which at a certain state invokes a service. That service needs to be aware of context changes as it contains an interval and based on a specific context value, it needs to respond accordingly. When we invoke the service like so:
```
invoke: {
  src: (context) => (callback) => captureEvidence(callback, 'videoCaptureComplete', null, context),
},
```
it has the initial values of the context but they never update. We update them with actions and the actions fire correctly updating the value. What is the right way to make a service aware of context changes?

**Comment by davidkpiano** [comment]
The current way is to give the invocation an `id` and then send events to it with the updated context. Imagine that the invoked actor lives in a completely different machine; it doesn't have the opportunity to "read" the context of the parent, which is only used for creation.

```js
invoke: {
  src: (context) => /* ... */,
  id: 'child'
},
on: {
  someEvent: {
    // transition that changes `context`
    actions: [
      sendTo('child', (ctx) => ({ type: 'parentUpdate', data: ctx }))
    ]
  }
}
```

I recommend:
- Only sending data that the invoked actor needs, instead of the entire `context`
- Only sending when needed; i.e. not on every transition

**Reply by Jonur** [reply]
Thanks for the quick response, @davidkpiano . Much appreciated! I will try it and let you know tomorrow.

**Reply by Gaccoo** [reply]
I have an app machine that spawns multiple other machines, like a database machine and a sync machine. Database machine initiates local db and stores it in context. Would this work the same way? If the sync machine wants to read a pending un-synced document in order o sync it. I should call an event to get the data from database? Should the database context be assigned to the app machine or the database machine?

**Reply by chandirag** [reply]
What's the recommended solution if the invoked actor is a promise Actor? According to the documentation (https://stately.ai/docs/promise-actors) it says that promise actors currently do not receive events?
